Adding data used by MAE and some bugfix

kent1D opened the pull request:
## Since this repository is now not designed only for one project, we add data used by mae:

* All missing ISO codes (American Samoa, Antarctica, Aruba, Aland, Caribbean Netherlands, Bouvet Island, Cocos Islands, Christmas Island, Western Sahara, French Guyana, Guadeloupe, Guam, Hong Kong, Heard Island and MacDonald, Macao, Northern Mariana Islands, Martinique, Nouvelle Calédonie, Norfolk Island, Polynésie Française, Saint Pierre et Miquelon, Puerto Rico, Réunion, Svalbard and Jan Mayen, Terres Australes Françaises, United States Minor Outlying Islands, Virgin Islands, U.S., Wallis et Futuna, Mayotte, Akrotiri et Dhekelia, Clipperton, Curaçao, Saint-Barthélemy, Saint Martin, Sint Maarten)
* `iso3`: ISO 3166-1 alpha-3
* `admin_level`: admin_level of the relation in OSM database
* `name:es`: spanish names
* `name:de`: german names
* `name:ru`: russian names
* `mae:region`: MAE specific regions of the world (comparable to
continents)
* `mae:circi_elec`: electoral circonscription of countries for french
people around the world
* `eu`: European Union (1 for yes, 0 for no)
* `sc`: small country (1 for yes, 0 for no)

## Fix some bugs or enhancements : 

* FIX #4 Only keep data properties from our csv file and not all datas from OSM
* FIX #5 Raise an error when overpass API returns a 429 HTTP error code before writing json file
* FIX #6 Indent exported files with 1 space

kent1D committed:
Since this repository is now not designed only for one project, we add
data used by mae :

* All missing ISO codes (American Samoa, Antarctica, Aruba, Aland,
Caribbean Netherlands, Bouvet Island, Cocos Islands, Christmas Island,
Western Sahara, French Guyana, Guadeloupe, Guam, Hong Kong, Heard Island
and MacDonald, Macao, Northern Mariana Islands, Martinique, Nouvelle
Calédonie, Norfolk Island, Polynésie Française, Saint Pierre et
Miquelon, Puerto Rico, Réunion, Svalbard and Jan Mayen, Terres Australes
Françaises, United States Minor Outlying Islands, Virgin Islands, U.S.,
Wallis et Futuna, Mayotte, Akrotiri et Dhekelia, Clipperton, Curaçao,
Saint-Barthélemy, Saint Martin, Sint Maarten)
* ISO 3166-1 alpha-3 as iso3 column
* admin_level: admin_level of the relation in OSM database
* name:es: spanish names
* name:de: german names
* name:ru: russian names
* mae:region: MAE specific regions of the world (comparable to
continents)
* mae:circi_elec: electoral circonscription of countries for french
people around the world
* eu: European Union (1 for yes, 0 for no)
* sc: small country (1 for yes, 0 for no)

kent1D committed:
* All missing ISO codes (American Samoa, Antarctica, Aruba, Aland,
Caribbean Netherlands, Bouvet Island, Cocos Islands, Christmas Island,
Western Sahara, French Guyana, Guadeloupe, Guam, Hong Kong, Heard Island
and MacDonald, Macao, Northern Mariana Islands, Martinique, Nouvelle
Calédonie, Norfolk Island, Polynésie Française, Saint Pierre et
Miquelon, Puerto Rico, Réunion, Svalbard and Jan Mayen, Terres Australes
Françaises, United States Minor Outlying Islands, Virgin Islands, U.S.,
Wallis et Futuna, Mayotte, Akrotiri et Dhekelia, Clipperton, Curaçao,
Saint-Barthélemy, Saint Martin, Sint Maarten)
* `iso3`: ISO 3166-1 alpha-3
* `admin_level`: admin_level of the relation in OSM database
* `name:es`: spanish names
* `name:de`: german names
* `name:ru`: russian names
* `mae:region`: MAE specific regions of the world (comparable to
continents)
* `mae:circi_elec`: electoral circonscription of countries for french
people around the world
* `eu`: European Union (1 for yes, 0 for no)
* `sc`: small country (1 for yes, 0 for no)

kent1D committed:
Since we play with different admin_level, we need to use it in the code.

We retrieve datas for iso codes with admin_level in 2 and 3

When admin_level is 0, the data can't be retrieved properly from
overpass (Antarctica, United States Minor Outlying Islands, Akrotiri et
Dhekelia, Heard Island and MacDonald, Western Sahara, Bouvet Island).

When admin_level is 4, the data is not so interesting to show on our map
then

kent1D committed:
Raise an error when overpass API returns a 429 HTTP error code before
writing json file

kent1D committed:
Only keep data properties from our csv file and not all datas from OSM

Should fix #4

kent1D committed:
Indent exported files with 1 space

Should fix #6

kent1D commented:
Seems like to be a big change in make.py but mainly for indentation inside a new test

yohanboniface commented on make.py:
```diff
@@ -131,49 +135,50 @@ def add_disputed(polygon, properties):
-            polygon = await add_area(conn, polygon, bir_tawil)
-        if iso == 'NP':
-            claim, props = await get_relation(conn, type="boundary",
+        if country['admin_level'] != '' and int(country['admin_level']) > 0  and int(country['admin_level']) < 4:
```
That may just be a naming issue, but having a `country` with `admin_level==''` seems awkward to me.
Do you have in mind how this can happen?

kent1D replied:
Antarctica doesn't have because it's not a country, or I didn't find it inside osm (tagging problem?)

But in any case this test `country['admin_level'] != ''` can be deleted I think, because I pushed 0 by default in the csv file

yohanboniface replied:
OK.
Maybe set the `admin_level` value once and for all before the `if`?
```python
admin_level = int(country['admin_level'] or 0)
if 0 < admin_level < 4:
    ….
```

kent1D replied:
Done in https://github.com/tilery/mae-boundaries/pull/7/commits/0b3656584beda08a02384bd6a2c30aeddfd3aa1c and https://github.com/tilery/mae-boundaries/pull/7/commits/3b882ebed8dd8afdc6d9b2304afccb8940cce63d

yohanboniface commented on make.py:
```diff
@@ -21,7 +21,11 @@
+            resp = requests.get(OVERPASS, params=params)
+            resp.raise_for_status()
+        except requests.exceptions.HTTPError as error:
+            raise ValueError(f'The request to overpass API failed: {error.response.status_code}')
```
Why not using `resp.status_code` directly here?

kent1D replied:
Because I am not familiar with this, and used an example found in documentation ;)

kent1D replied:
I changed it a bit following https://stackoverflow.com/questions/16511337/correct-way-to-try-except-using-python-requests-module, what do you think of: https://github.com/tilery/mae-boundaries/pull/7/commits/0b3656584beda08a02384bd6a2c30aeddfd3aa1c

But I think we should retry it after a while if we have a 429 HTTP error if we want to lauch this script with CI.

Maybe in an other enhancement report?

kent1D committed:
Better handling of requests error :

- if no connection
- if the API of overpass returns a HTTP Error (429 in general)

We fail silently without traceback

kent1D committed:
Being sure we have an admin_level

yohanboniface commented on make.py:
```diff
@@ -45,7 +54,7 @@
         collection.append(LineString(coords))
     shape = await make_polygon(conn, MultiLineString(collection))
-    return shape, relation.tags
+    return shape, {}
```
Why returning a tuple in the end now? Seems like we could return only the `shape`, no?

kent1D replied:
Should be ok now : https://github.com/tilery/mae-boundaries/pull/7/commits/f14bdc95bb3eb583f3cfd64e44f896d424940a51

kent1D committed:
Better error handling (printing)

There is no more second element in the return of the function
get_relation()

kent1D commented:
ping @yohanboniface some more modifications needed?
